langchain4j-cassandra/src/test/java/dev/langchain4j/store/embedding/cassandra/SampleDocumentLoaderAndRagWithAstraTest.java

```diff
@@ -0,0 +1,142 @@
+package dev.langchain4j.store.embedding.cassandra;
+
+import com.dtsx.astra.sdk.utils.TestUtils;
+import dev.langchain4j.data.document.Document;
+import dev.langchain4j.data.document.DocumentSplitter;
+import dev.langchain4j.data.document.FileSystemDocumentLoader;
+import dev.langchain4j.data.document.splitter.DocumentSplitters;
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.model.input.Prompt;
+import dev.langchain4j.model.input.PromptTemplate;
+import dev.langchain4j.model.openai.OpenAiChatModel;
+import dev.langchain4j.model.openai.OpenAiEmbeddingModel;
+import dev.langchain4j.model.openai.OpenAiTokenizer;
+import dev.langchain4j.store.embedding.EmbeddingMatch;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import dev.langchain4j.store.embedding.EmbeddingStoreIngestor;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static com.dtsx.astra.sdk.utils.TestUtils.readToken;
+import static com.dtsx.astra.sdk.utils.TestUtils.setupDatabase;
+import static dev.langchain4j.model.openai.OpenAiModelName.GPT_3_5_TURBO;
+import static dev.langchain4j.model.openai.OpenAiModelName.TEXT_EMBEDDING_ADA_002;
+import static java.time.Duration.ofSeconds;
+import static java.util.stream.Collectors.joining;
+
+public class SampleDocumentLoaderAndRagWithAstraTest {
+
+    @Test
+    @Disabled("To run you need both OpenAi and Astra keys")
+    public void shouldRagWithOpenAiAndAstra() throws InterruptedException {
+        // Initialization
+        String astraToken  = readToken();
+        String databaseId  = setupDatabase("langchain4j", "langchain4j");
+        String openAIKey   = System.getenv("OPENAI_API_KEY");
+
+        // Given
+        Assertions.assertNotNull(openAIKey);
+        Assertions.assertNotNull(databaseId);
+        Assertions.assertNotNull(astraToken);
+
+        // --- Ingesting documents ---
+
+        // Parsing input file
+        Document document = FileSystemDocumentLoader
+                .loadDocument(Objects.requireNonNull(SampleDocumentLoaderAndRagWithAstraTest.class
```

Comment:
this test is failing when running via IDE:

`java.nio.file.InvalidPathException: Illegal char <:> at index 2: /C:/dev/repo/langchain4j-clun/langchain4j-cassandra/target/test-classes/story-about-happy-carrot.txt`

You can use similar approach to `FileSystemDocumentLoaderTest.toPath()` to load file, that will work when running in IDE and via maven

Reply:
All done wooo